lightning/src/ln/functional_tests.rs

```diff
@@ -10511,6 +10511,90 @@ fn test_remove_expired_inbound_unfunded_channels() {
 	check_closed_event(&nodes[1], 1, ClosureReason::HolderForceClosed, false, &[nodes[0].node.get_our_node_id()], 100000);
 }
 
+#[test]
+	fn test_channel_close_when_not_timely_accepted() {
+		// Create network of two nodes
+		let chanmon_cfgs = create_chanmon_cfgs(2);
+		let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+		let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+		let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+		// Simulate peer-diconnects mid-handshake
+		// The channel is initiated from the node 0 side,
+		// But the nodes disconnects before node 1 could send accept channel
+		let create_chan_id = nodes[0].node.create_channel(nodes[1].node.get_our_node_id(), 100000, 10001, 42, None, None).unwrap();
+		let open_channel_msg = get_event_msg!(nodes[0], MessageSendEvent::SendOpenChannel, nodes[1].node.get_our_node_id());
+		assert_eq!(open_channel_msg.temporary_channel_id, create_chan_id);
+
+		nodes[0].node.peer_disconnected(&nodes[1].node.get_our_node_id());
+		nodes[1].node.peer_disconnected(&nodes[0].node.get_our_node_id());
+
+		{
+			// Make sure that we have not removed the OutboundV1Channel from node[0] immediately.
+			let node_0_per_peer_state = nodes[0].node.per_peer_state.read().unwrap();
+			let per_peer_state = node_0_per_peer_state.get(&nodes[1].node.get_our_node_id()).unwrap().lock().unwrap();
+			assert_eq!(per_peer_state.channel_by_id.len(), 1);
```

Comment:
Should we also check that `nodes[1]` _does_ drop the peer?

Reply:
I think that's a great idea!
Updated the test in 15792af